spec/index.bs

```diff
@@ -5131,70 +5144,114 @@ enum GPUStoreOp {
 <dl dfn-type=method dfn-for=GPURenderEncoderBase>
     : <dfn>setPipeline(pipeline)</dfn>
     ::
+        Sets the current {{GPURenderPipeline}}.
 
         <div algorithm="GPURenderEncoderBase.setPipeline">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setPipeline(pipeline)">
-                pipeline:
+                |pipeline|: The render pipeline to use for subsequent drawing commands.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setPipeline()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, generate a
+                    {{GPUValidationError}} error and stop.
+                    <div class=validusage>
+                        - |pipeline| is [=valid=].
+                        - |pipeline|.{{GPUObjectBase/[[device]]}} is |this|.
+                    </div>
+                1. Set |this|.{{GPURenderEncoderBase/[[pipeline]]}} to be |pipeline|.
+            </div>
         </div>
 
     : <dfn>setIndexBuffer(buffer, indexFormat, offset, size)</dfn>
     ::
+        Sets the current index buffer.
 
         <div algorithm="GPURenderEncoderBase.setIndexBuffer">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setIndexBuffer(buffer, indexFormat, offset, size)">
-                buffer:
-                indexFormat:
-                offset:
-                size:
+                |buffer|: Buffer containing index data to use for subsequent drawing commands.
+                indexFormat: Format of the index data contained in |buffer|.
+                |offset|: Offset in bytes into |buffer| where the index data begins.
+                |size|: Size in bytes of the index data in |buffer|.
+                    If `0`, |buffer|.{{GPUBuffer/[[size]]}} - |offset| is used.
```

Comment:
We've been going through the api recently and removing some sentinel values. We should change this so this arg has no default value, and have it only default if it's unspecified. Same in setVertexBuffer.

API change, so could be a followup.

Reply:
Yeah, I'd prefer to leave this till an API change is agreed on.